interop-reactiveStreams/jvm/src/test/scala/scalaz/zio/interop/reactiveStreams/QueueSubscriberSpec.scala

```diff
@@ -35,82 +35,87 @@ class QueueSubscriberSpec(implicit ee: ExecutionEnv) extends TestRuntime {
-    ) should_=== Success(())
+        _  <- UIO(subscriber.onSubscribe(s))
+        _  <- fiber.await
+        ro <- canceled.poll
+        r  <- ro.fold(UIO.succeed[Result](failure))(identity)
+      } yield r
+    )
 
   private def e3 =
-    unsafeRunSync(
-      (for {
+    unsafeRun(
+      for {
         subStr               <- QueueSubscriber.make[Int](10)
         (subscriber, stream) = subStr
-        canceled             <- Promise.make[Unit, Unit]
+        canceled             <- Promise.make[Nothing, Result]
         runtime              <- ZIO.runtime[Any]
         s = new Subscription {
-          override def request(n: Long): Unit = runtime.unsafeRun(canceled.fail(()).unit)
-          override def cancel(): Unit         = runtime.unsafeRun(canceled.succeed(()).unit)
+          override def request(n: Long): Unit = runtime.unsafeRun(canceled.succeed(failure).unit)
+          override def cancel(): Unit         = runtime.unsafeRun(canceled.succeed(success).unit)
         }
-        _ <- UIO(subscriber.onSubscribe(s))
-        _ <- stream.run(Sink.fail(boom)).catchAll(_ => UIO.unit)
-        _ <- canceled.await
-      } yield ()).timeoutFail(())(Duration(2, SECONDS))
-    ) should_=== Success(())
+        _  <- UIO(subscriber.onSubscribe(s))
+        _  <- stream.run(Sink.fail(boom)).catchAll(_ => UIO.unit)
+        ro <- canceled.poll
+        r  <- ro.fold(UIO.succeed[Result](failure))(identity)
+      } yield r
+    )
 
   private def e4 =
-    unsafeRunSync(
-      (for {
+    unsafeRun(
+      for {
         subStr               <- QueueSubscriber.make[Int](10)
         (subscriber, stream) = subStr
-        canceled             <- Promise.make[Unit, Unit]
+        canceled             <- Promise.make[Nothing, Result]
         runtime              <- ZIO.runtime[Any]
         s = new Subscription {
           override def request(n: Long): Unit = (0 until n.toInt).foreach(subscriber.onNext)
-          override def cancel(): Unit         = runtime.unsafeRun(canceled.succeed(()).unit)
+          override def cancel(): Unit         = runtime.unsafeRun(canceled.succeed(success).unit)
         }
-        _ <- UIO(subscriber.onSubscribe(s))
-        _ <- stream.drop(10).run(Sink.fail(boom)).catchAll(_ => UIO.unit)
-        _ <- canceled.await
-      } yield ()).timeoutFail(())(Duration(2, SECONDS))
-    ) should_=== Success(())
+        _  <- UIO(subscriber.onSubscribe(s))
+        _  <- stream.drop(10).run(Sink.fail(boom)).catchAll(_ => UIO.unit)
+        ro <- canceled.poll
+        r  <- ro.fold(UIO.succeed[Result](failure))(identity)
+      } yield r
+    )
 
   private def e5 =
-    unsafeRunSync(
-      (for {
+    unsafeRun(
+      for {
         subStr               <- QueueSubscriber.make[Int](10)
         (subscriber, stream) = subStr
         fiber                <- stream.run(Sink.collect[Int]).fork
         _                    <- ZIO.sleep(Duration(100, MILLISECONDS))
```

Comment:
Do we still need the `sleep` here? If you're trying to make sure the fiber actually starts running, a common trick is using a `Promise` that is resolved by the fiber before it runs and awaiting on it before progressing.